resources/splunk_app.rb

```diff
@@ -136,3 +148,41 @@ def config_scope
     apply_config
   end
 end
+
+class PackagedApp < SplunkApp
+  property :source_url, String, required: true
+
+  resource_name :splunk_app_package
+
+  def package_path
+    return @package_path if @package_path
+    file_cache = Pathname.new(Chef::Config['file_cache_path'])
+    @package_path = file_cache.join(CernerSplunk::PathHelpers.filename_from_url(source_url).gsub(/.spl$/, '.tgz'))
+  end
+
+  action :install do
+    return unless !version || changed?(:version)
+
+    remote_file package_path.to_s do
```

Comment:
There is a [check](https://github.com/cerner/cerner_splunk/blob/stable/libraries/splunk_app.rb#L199) for snapshot version installing over released version before downloading the file in the current splunk_app implementation in cerner_splunk. Also, we do some [validation](https://github.com/cerner/cerner_splunk/blob/stable/libraries/splunk_app.rb#L223-L239) over the downloaded tarball before installing it. Should we be including those checks?